src/sfdxError.ts

```diff
@@ -205,6 +205,11 @@ export class SfdxError extends NamedError {
       sfdxError.stack = sfdxError.stack.replace(`${err.name}: ${err.message}`, 'Outer stack:');
       sfdxError.stack = `${err.stack}\n${sfdxError.stack}`;
     }
+
+    // If the original error has a code, use that instead of name.
+    if (hasString(err, 'code')) {
+      sfdxError.code = ensureString(getString(err, 'code'));
```

Comment:
No need to change this here, but when you really only need one of the type-specific fns here when "ensuring".  e.g. `ensure(getString(err, 'code'))` or `ensureString(get(err, 'code'))`.  I tend to prefer using the untyped `ensure` style when the type is inferrable from the arg.

Reply:
Oh actually can make this much simpler since `hasString` will re-type `err`: it can be `sfdxError.code = err.code`.

Reply:
That does work! I could have swore I tried that.